core/file_unix.cpp

```diff
@@ -154,11 +151,22 @@ bool File::canonical(const std::string& path, std::string* resolved) {
   free(result);
   return true;
 }
+
 long File::freespace_for_path(const string& path) {
+#if defined(__sun)
```

Comment:
Looks like statvfs is POSIX, and also on linux.  I should just use that one. once I double check it works on linux.  Then this #if defined can go away too